Marlin/src/module/configuration_store.cpp

```diff
@@ -1906,7 +1922,10 @@ void MarlinSettings::reset(PORTARG_SOLO) {
   #if HAS_BED_PROBE
     zprobe_zoffset = Z_PROBE_OFFSET_FROM_EXTRUDER;
   #endif
-
+  #if ENABLED(INDEPENDENT_Z_OFFSETS) && ENABLED(DUAL_X_CARRIAGE)
+    x1_zprobe_zoffset = zprobe_zoffset;
+    x2_zprobe_zoffset = zprobe_zoffset;
```

Comment:
Instead of using two variables, it would be better to have a single two-element array, `float zprobe_zoffset[2]`.

Reply:
Makes sense.

I'm still not very clear how two probes is going to work. You have such a setup?

Reply:
Just one probe, but the nozzles are physically at different heights, therefore each nozzle needs a different offset from the same probe.

Reply:
The Z probe Z offset is only used during probing. And we already know the second nozzle's Z offset from `hotend_offset[Z_AXIS]`. If there is no second probe attached to the second nozzle, then there's nothing useful being added by a second Z probe Z offset.